### PR TITLE
DLUHC-201 add timetable form skeleton

### PR DESCRIPTION
**Pull request by samornsby**
Add timetable form skeleton and basic pages for title, description and publish date.
- Added Timetable form component
- Added pages for title, description and publish date
- Added custom inputs for a textarea and date
- Added a story to display the form

Minimal styling has been added, the rest will be added in future PRs, UX can be found here under SOW14: https://www.figma.com/file/XaBQCco5EI92PaBDXq7Mqq/DLUHC-New-Local-Plans-timetable-dashboard-concepts?type=design&node-id=0-1&mode=design
The stages is also quite a large chunk of work so will be added in another PR
image
image
image

### CODE REVIEW

**Review by bethpritchard — APPROVED** [review]
Nice, matches figma (without obvious styling bits) - way slicker than the prototype kit version!

File: dluhc-component-library/src/components/timetableForm/TimetableForm.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { useMemo, useState } from "preact/hooks";
+import DescriptionPage from "./components/DescriptionPage";
+import PublishedDatePage from "./components/PublishedDatePage";
+import TitlePage from "./components/TitlePage";
+
+const TIMETABLE_FORM_PAGES = new Map([
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/components/timetableForm/TimetableForm.tsx:6]: I didnt think about actually using a `Map()` i was thinking an object, however i think Map keeps the ordering though so we may not need the `TIMETABLE_PAGE_IDS` variable. Worth looking into

File: dluhc-component-library/src/components/timetableForm/TimetableForm.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  ["title", TitlePage],
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/components/timetableForm/TimetableForm.tsx:7]: If we do keep both variables for the pages, you should pull `title`, `description` and `publishedDate` out into constants so they always match.

File: dluhc-component-library/src/components/timetableForm/components/DateInput.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const DateInput = () => {
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/components/timetableForm/components/DateInput.tsx:1]: `Non-blocker` Now that we have more than one form, we should probably pull the none specific sub components to the normal component level. Eg `Textarea` and `DateInput` will probably be useful in both.

File: dluhc-component-library/src/components/timetableForm/TimetableForm.tsx (hunk 1 above)
Comment (samornsby) [reply dluhc-component-library/src/components/timetableForm/TimetableForm.tsx]: As discussed offline, the components for stages will need props to be passed in. My current implementation is to have a list of objects where the object has the id and the component. Once I add stages I'll put the props on those objects.

File: dluhc-component-library/src/components/timetableForm/TimetableForm.tsx (hunk 2 above)
Comment (samornsby) [reply dluhc-component-library/src/components/timetableForm/TimetableForm.tsx]: I think this isn't needed with the way I've done it as a list now, but if that changes and we do have both variables I'll make this change.

File: dluhc-component-library/src/components/timetableForm/components/DateInput.tsx (hunk 3 above)
Comment (samornsby) [reply dluhc-component-library/src/components/timetableForm/components/DateInput.tsx]: I've moved `DateInput` and `Textarea` into new folders: `/components/dateInput` and `/components/textarea`

**Review by amjones-scottlogic — APPROVED** [review]
LGTM

**Pull request merged**